uber/models.py

```diff
@@ -630,9 +630,22 @@ def everything(self, location=None):
                                         .order_by(Attendee.full_name).all()
                          if c.AT_THE_CON or not location or int(location) in a.assigned_depts_ints]
             for job in jobs:
-                job._available_staffers = [a for a in attendees if not job.restricted or a.trusted]
+                job._available_staffers = [a for a in attendees if not job.restricted or a.trusted_in(job.location)]
             return jobs, shifts, attendees
 
+        def staffers_by_job_options(self, job):
+            """
+            Format staffers_by_job() for use with the {% options %} template decorator
+            """
+            return [(a.id, a.full_name) for a in self.staffers_by_job(job)]
+
+        def staffers_by_job(self, job):
+            return self.query(Attendee)\
+                .filter_by(staffing=True)\
+                .filter(Attendee.assigned_depts.contains(str(job.location)))\
+                .filter(*(Attendee.trusted_depts.contains(str(job.location)),) if job.restricted else ())\
+                .order_by(Attendee.full_name).all()
+
         def search(self, text, *filters):
```

Comment:
The reason for the difference is performance; a lot of these functions are simple-but-inefficient, so we ended up tacking on some more-efficient versions without ever getting rid of the older versions.  Which is not idea, and there is indeed a backlog ticket for cleaning all that up and making sure it performs well, etc.